theano/gof/destroyhandler.py

```diff
@@ -13,6 +13,7 @@
 from theano.misc.ordered_set import OrderedSet
 
 from .fg import InconsistencyError
+from Queue import Queue
```

Comment:
This is your problem.  It has been renamed in python3.  Use `from six.moves.queue import Queue` instead.